Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridComboBoxColumn.cs

```diff
@@ -0,0 +1,606 @@
+                if (item != null)
+                {
+                    var newValue = DisplayMemberPath != null
+                        ? item.GetType().GetProperty(Binding.Path.Path).GetValue(item)
+                        : item;
+
+                    if (dataItem != null)
+                    {
+                        dataItem.GetType().GetProperty(Binding.Path.Path).SetValue(dataItem, newValue);
+                    }
+                    else
+                    {
+                        var dataType = this.OwningGrid.ItemsSource.GetItemType();
+                        var newDataItem = Activator.CreateInstance(dataType);
+                        dataType.GetProperty(Binding.Path.Path).SetValue(newDataItem, newValue);
+                        dataItem = newDataItem;
+                    }
+                }
+            };
+
+            return comboBox;
+        }
+
+        /// <summary>
+        /// Gets a read-only <see cref="T:Windows.UI.Xaml.Controls.TextBlock"/> element that is bound to the column's <see cref="P:Microsoft.Toolkit.Uwp.UI.Controls.DataGridBoundColumn.Binding"/> property value.
+        /// </summary>
+        /// <param name="cell">The cell that will contain the generated element.</param>
+        /// <param name="dataItem">The data item represented by the row that contains the intended cell.</param>
+        /// <returns>A new, read-only <see cref="T:Windows.UI.Xaml.Controls.TextBlock"/> element that is bound to the column's <see cref="P:Microsoft.Toolkit.Uwp.UI.Controls.DataGridBoundColumn.Binding"/> property value.</returns>
+        protected override FrameworkElement GenerateElement(DataGridCell cell, object dataItem)
+        {
+            EnsureColumnBinding(dataItem);
+
+            EnsureDisplayMemberPathExists();
+
+            var textBlockElement = new TextBlock
+            {
+                Margin = new Thickness(DATAGRIDCOMBOBOXCOLUMN_leftMargin, 0.0, DATAGRIDCOMBOBOXCOLUMN_rightMargin, 0.0),
+                VerticalAlignment = VerticalAlignment.Center
+            };
+
+            if (DependencyProperty.UnsetValue != ReadLocalValue(DataGridComboBoxColumn.FontFamilyProperty))
+            {
+                textBlockElement.FontFamily = this.FontFamily;
+            }
+
+            if (_fontSize.HasValue)
+            {
+                textBlockElement.FontSize = _fontSize.Value;
+            }
+
+            if (_fontStyle.HasValue)
+            {
+                textBlockElement.FontStyle = _fontStyle.Value;
+            }
+
+            if (_fontWeight.HasValue)
+            {
+                textBlockElement.FontWeight = _fontWeight.Value;
+            }
+
+            RefreshForeground(textBlockElement, (cell != null & cell.OwningRow != null) ? cell.OwningRow.ComputedForeground : null);
+
+            if (this.Binding != null && EnsureOwningGrid())
+            {
+                if (string.IsNullOrEmpty(DisplayMemberPath))
```

Comment:
I had to add the 'else' statement:

if (string.IsNullOrEmpty(DisplayMemberPath)) 
{ 
    textBlockElement.SetBinding(TextBlock.TextProperty, this.Binding); 
}
else
{
    textBlockElement.Text = GetDisplayValue(dataItem);
}

This was to avoid getting blank cells when adding a ComboBox column to an already populated DataGrid.

After those three changes, things started to work better.   Note that I will be out of the office tomorrow.

Thanks for the contribution!

Reply:
@MSRegisB  I never tested this case so thank you for that! I will add the else statement.

My pleasure and thanks for the thorough review. Enjoy your weekend!